src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -39,6 +39,37 @@ static const std::set<std::wstring_view> ProfileGeneratorNamespaces{
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
+    bool ProfilePageNavigationState::CanDeleteProfile() const
+    {
+        const auto guid{ Profile().Guid() };
+        const std::wstring_view source{ Profile().Source() };
+        if (IsBaseLayer())
+        {
+            return false;
+        }
+        else if (InBoxProfileGuids.find(guid) != InBoxProfileGuids.end())
+        {
+            // in-box profile
+            return false;
+        }
+        else if (LegacyDynamicProfileGuids.find(guid) != LegacyDynamicProfileGuids.end() ||
+                 ProfileGeneratorNamespaces.find(source) != ProfileGeneratorNamespaces.end())
```

Comment:
why not just "has a source"? That covers all of these cases without us having to hardcode a list of sources.

Reply:
Changing to "has a source". Curious, what's the point of the legacy dynamic profile guids again? I ran into them and thought they were important for a situation like this.